communications/ff_msgs/action/Plan.action

```diff
@@ -28,7 +28,7 @@ float32 desired_omega                         # Desired (max) omega
 float32 desired_alpha                         # Desired (max) alpha
 float32 desired_rate                          # Desired rate
 
-duration max_time                             # Max generation time
+builtin_interfaces/Duration max_time                             # Max generation time
```

Comment:
I don't think this version will compile in ROS1, so we would need a way to specify different field types for different ROS versions.

This becomes an example of where we might want to use the symlinking script, naming this file `jinja-Plan.action` and writing it:
```jinja
{% if USE_ROS1 %}
duration max_time
{% else %}
builtin_interfaces/Duration max_time
{% endif %}
```

Or maybe there is some other way to address it. If the build system doesn't require us to use the canonical folder and filename for each action file, I guess it can be handled by including different `Plan.action` files (perhaps in `ros1` and `ros2` subfolders) in the conditional branches of the `CMakeLists.txt`.

Reply:
We all would like to make the ros2 branch ros1 compatible, but I'm not entirely convinced it will be feasible at a reasonable level of effort. I think we left it that we needed to better understand ROS2 and convert a first package before we would decide. Let's discuss tomorrow.

Reply:
I think to make it compatible we need to go the template way, I'm not opposing it but am afraid of the extra effort needed

Reply:
A symlinking script that also implements templates is available at [1], tested and with usage instructions at the top. Probably it can be better tuned to agree with people's desired conventions and what we've learned since it was written. I don't insist that we use it and would be thrilled if we can find simpler approaches that make it unnecessary, but I am surprised to hear that it's not feasible!

[1] https://github.com/trey0/astrobee/blob/c4cafd764ddb9d80d4d5d52bb5cf6eacbb4f2d85/scripts/build/autogen_ros_version_src.py

Reply:
I'll give this a try and maybe bring some inputs by tomorrow's meeting @trey0

Reply:
Feasible *at a reasonable level of effort*. I'm sure we can do it, but if it takes twice as much work as managing two branches, I don't think it's worth it.